pkgs/development/python-modules/importlib-metadata/default.nix

```diff
@@ -1,45 +1,35 @@
 { lib
 , buildPythonPackage
 , fetchPypi
-, setuptools_scm
-, zipp
-, pathlib2
-, contextlib2
-, configparser
-, isPy3k
-, importlib-resources
+, pythonOlder
+, setuptools-scm
 , toml
-, packaging
+, zipp
 }:
 
 buildPythonPackage rec {
   pname = "importlib-metadata";
   version = "3.4.0";
+  disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     pname = "importlib_metadata";
     inherit version;
     sha256 = "sha256-+l2qRHenQUrjTpWULk3Qf2Kt9YkUPIdcEzweU8Tv840=";
   };
 
-  nativeBuildInputs = [ setuptools_scm ];
-
-  propagatedBuildInputs = [ toml zipp ]
-    ++ lib.optionals (!isPy3k) [ pathlib2 contextlib2 configparser ];
-
-  doCheck = false; # Cyclic dependencies.
+  nativeBuildInputs = [ setuptools-scm ];
 
-  # removing test_main.py - it requires 'pyflakefs'
-  # and adding `pyflakefs` to `checkInputs` causes infinite recursion.
-  preCheck = ''
-    rm importlib_metadata/tests/test_main.py
-  '';
+  propagatedBuildInputs = [ toml zipp ];
 
+  # Cyclic dependencies due to pyflakefs
```

Comment:
I think we should really execute tests here. Can we use something like https://github.com/NixOS/nixpkgs/pull/113088/files#diff-7c765cc9768cdae049f8712b4ef719bc94ec8ec2e8df2dd50ddf14c6acd4e580R3069 to avoid the cyclic dependency?

Reply:
I'm not able to do this or I can't wrap my head around it. `importlib-metadata` is not a implicit dependency of `pyflakefs` thus the overwriting to not run the tests sees to fail as it seems.